azure-pipelines.yml

```diff
@@ -1,89 +1,179 @@
+    steps:
+
+      - script: |
+          npm install
+        displayName: 'npm install'
+
+      - script: |
+          npm run build
+        env:
+          REACT_APP_CLIENT_ID: $(REACT_APP_STAGING_CLIENT_ID)
+          REACT_APP_INSTRUMENTATION_KEY: $(REACT_APP_INSTRUMENTATION_KEY)
+        displayName: 'Build static assets for staging'
+
+      - task: PowerShell@2
+        displayName: 'Set version-number'
+        condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+        inputs:
+          targetType: 'inline'
+          script: |
+            $lernaPath = "$Env:BUILD_SOURCESDIRECTORY\package.json"
+            $json = Get-Content "$lernaPath" | Out-String | ConvertFrom-Json
+            $versionNumber = $json.version;
+            Write-Host "Setting the build number variable to '$versionNumber'."
+            Write-Host "##vso[build.updatebuildnumber]$versionNumber"
+
+      - script: |
+          npm run build
+        condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+        env:
+          REACT_APP_CLIENT_ID: $(REACT_APP_PROD_CLIENT_ID)
+          REACT_APP_INSTRUMENTATION_KEY: $(REACT_APP_INSTRUMENTATION_KEY)
+        displayName: 'Build static assets for prod'
+
+      - task: CopyFiles@1
+        displayName: 'Copy files to: $(Build.ArtifactStagingDirectory)'
+        inputs:
+          SourceFolder: '$(System.DefaultWorkingDirectory)/build'
+          TargetFolder: '$(Build.ArtifactStagingDirectory)'
+
+      - task: PublishBuildArtifacts@1
+        displayName: 'Publish Artifact: drop'
+        inputs:
+          PathtoPublish: '$(build.ArtifactStagingDirectory)'
+
+      - task: DeleteFiles@1
+        inputs:
+          contents: node_modules
+        displayName: 'Delete node_modules'
+
+      - task: AzureRmWebAppDeployment@3
+        condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/dev'))
+        inputs:
+          azureSubscription: 'Graph-Explorer-ARM'
+          WebAppName: 'graphTryIt-staging'
+          Package: '$(System.DefaultWorkingDirectory)'
+          GenerateWebConfig: true
+          WebConfigParameters: '-Handler iisnode -NodeStartFile server.js -appType node'
+        displayName: 'Deploy to staging'
+        enabled: false
+
+      - task: AzureFileCopy@2
+        inputs:
+          sourcePath: 'build'
+          azureConnectionType: 'ConnectedServiceNameARM'
+          azureSubscription: 'arm-connection'
+          destination: 'azureBlob'
+          storage: 'graphstagingblobstorage'
+          containerName: 'staging/vendor/bower_components/explorer-v2/build'
+          cleanTargetBeforeCopy: true
+        condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/dev'))
+        displayName: 'Deploy to -tst'
+        enabled: false
+
+      - task: AzureRmWebAppDeployment@3
+        condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
```

Comment:
With the successful deployment to tst, you can now remove these disabled tasks. Still keep the notification disabled as we still need to create that connection.